### PR TITLE
Create DeleteLanguage Button

### LINKED ISSUE #90: Language Editor: delete language
The user should be able to delete a language by clicking the `Delete Language` button.

- [x] add a `Delete Language` button to the Language Editor (after all the other fields)
- [x] when the user clicks the button, prompt them to confirm by typing `YES` (use `window.prompt()`)
- [x] if they confirm, add a `deleted: true` property to the language and save it, then set the `app.settings.language` property to `null`, then navigate to `/languages` (which will trigger the app to rerender, displaying the Languages page, but with no language set)
- [x] update any code that loads the list of languages to load only those languages that are _not_ marked as deleted (might want to add a `db.getLanguages()` method to encapsulate this functionality / make it DRY)
- [x] button should use the button styles from the [DLx style library](https://styles.digitallinguistics.io/section-components.html); since we'll be using button styles in tons of components, let's include the button style in `App.less`
- [x] button should be colored red (we'll need to add a red color to our palette; this should probably be done in the styles library first; see https://github.com/digitallinguistics/styles/issues/144)

**Confirmation Prompt**
> Are you sure you want to delete this Language? The data can be recovered at any time.

### LINKED ISSUE #253: update DLx JavaScript library to v0.3.0
Specifically, we need a version of the library which includes `iso` and `glottolog` properties on the `Language` model.

- [x] Update DLx JavaScript library
- [x] Remove ISO Code code in `src/models/Language.js`

This will allow the testing data to be formatted correctly.

### PR DESCRIPTION
**Pull request by vadekh**
**Related Issue:**
closes #90
closes #253

## Description
Creates a delete button in the language editor, which prompts the user, and then 'deletes' a language, essentially removing it from view and rerendering the elements that referenced it.

## Checklist

- [x] Check out the [Contributing Guidelines][contributing] if you need help getting started with your pull request.

- [x] [Open an issue][issues] for the change (if one doesn't already exist).

- [x] Update tests for planned changes. Check that they are failing. (`npm test`)

- [x] Write code to pass the tests, then build the project. (`npm run build`)

- [x] Check that tests are now passing (`npm test`). This will also build the project again.

- [x] Add inline code commenting (using [JSDoc][JSDoc] style).

- [x] Generate the developer documentation (`npm run docs`) and check your changes by opening `docs/index.html` in a browser.

- [x] If you're adding a new component, add yourself as a [codeowner][codeowners] for that component.

[codeowners]:   https://github.com/digitallinguistics/app/blob/master/.github/CODEOWNERS
[contributing]: https://github.com/digitallinguistics/app/blob/master/.github/CONTRIBUTING.md
[issues]:       https://github.com/digitallinguistics/app/issues
[JSDoc]:        https://jsdoc.app/
[Node]:         https://nodejs.org/en/
[nvm]:          https://github.com/creationix/nvm
[nvm-windows]:  https://github.com/coreybutler/nvm-windows

### CODE REVIEW

**Comment by vadekh** [comment]
@dwhieb The failing test is passing on my machine. Do we have to resolve it before merging? Also it passed on the run tests 1-4 I think if I understand how this works correctly.

**Comment by dwhieb** [comment]
I suspect this is a race condition. We might need to add a statement to the test that waits for an element to be rendered before the next part of the test runs.

There's a section in this blog post about calling too many actions too fast:

https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/

**Comment by vadekh** [comment]
@dwhieb Ah, so like the tests in languageDropdown that have `cy.get(.language-editor);` before the test conditions are checked?

**Comment by dwhieb** [comment]
Bingo

File: src/app/App.js [hunk 1]
```diff
@@ -203,7 +202,19 @@ class App extends View {
     return language;
   }
 
-
+  async deleteLanguage(idToDelete) {
+    const confirm = window.prompt(`Are you sure you want to delete this Language? The data can be recovered at any time.`);
```
Comment (dwhieb) [inline src/app/App.js:206]:
```suggestion
    const confirmed = window.prompt(`Are you sure you want to delete this Language? The data can be recovered at any time. Type "YES" to delete.`);
```

There's a `window.confirm` property, so using `confirm` as the variable name here shadows that property, and might cause confusion down the road.

The user also doesn't know what they need to type based on the original message, so I added the last sentence to the prompt.

File: src/components/LanguageEditor/LanguageEditor.html [hunk 2]
```diff
@@ -42,4 +42,6 @@
     </div>
   </label>
 
+  <button type='button' class='button red' id=delete-language-button>Delete this Language</button>
```
Comment (dwhieb) [inline src/components/LanguageEditor/LanguageEditor.html:45]:
```suggestion
  <button type=button class='button red' id=delete-language-button>Delete this Language</button>
```

File: src/app/App.js [hunk 3]
```diff
@@ -203,7 +202,19 @@ class App extends View {
...
+    if (confirm === 'YES') {
```
Comment (dwhieb) [inline src/app/App.js:207]:
```suggestion
    if (confirmed === 'YES') {
```

File: src/components/LanguageEditor/LanguageEditor.test.js [hunk 4]
```diff
@@ -34,6 +38,35 @@ describe(`LanguageEditor`, () => {
...
+      // Check for Delete Button
+      cy.get(`#delete-language-button`)
+      .should(`be.visible`);
+    });
+
+  });
+
+  it(`deletes language with button`, () => {
+    cy.window().then(win => {
+      cy.stub(win, 'prompt').returns(`YES`);
```
Comment (dwhieb) [inline src/components/LanguageEditor/LanguageEditor.test.js:50]: Nice! Now I know how to do this.

File: src/components/LanguageEditor/LanguageEditor.test.js [hunk 5]
```diff
@@ -34,6 +38,35 @@ describe(`LanguageEditor`, () => {
...
       .should(`have.value`, `stan1288`);
 
+      // Check for Delete Button
+      cy.get(`#delete-language-button`)
+      .should(`be.visible`);
+    });
+
+  });
+
+  it(`deletes language with button`, () => {
```
Comment (dwhieb) [inline src/components/LanguageEditor/LanguageEditor.test.js:48]: Great set of tests for this.

**Pull request merged**